adjust default behavior of text and display style bracket-like commands

jhuggins opened the pull request:
Addresses issue raised by @jnegrea in #1

jnegrea commented on shortex.sty:
```diff
@@ -146,17 +149,17 @@
+\newcommand{\cbra}[2][{-2}]{\lcrxx[#1] \{ {#2} \} }
+\newcommand{\sbra}[2][{-2}]{\lcrxx[#1] [ {#2} ] }
+
+\newcommand{\abs}[2][{-1}]{\lcrxx[#1] \vert {#2} \vert }
```
Bracket-like notation that was defined by lcrx before and is defined by lcrxx now has default argument -1 instead of -2 (\abs ... \card lines 152-162). 
Is this on purpose?
I think they should use -2 to have the same behaviour \rbra, \cbra,...

jnegrea commented on shortex.sty:
```diff
@@ -146,17 +149,17 @@
+\newcommand{\cbra}[2][{-2}]{\lcrxx[#1] \{ {#2} \} }
+\newcommand{\sbra}[2][{-2}]{\lcrxx[#1] [ {#2} ] }
+
+\newcommand{\abs}[2][{-1}]{\lcrxx[#1] \vert {#2} \vert }
+\newcommand{\set}[2][{-1}]{\lcrxx[#1] \{ {#2} \}}
+\newcommand{\floor}[2][{-1}]{\lcrxx[#1] \lfloor {#2} \rfloor}
+\newcommand{\ceil}[2][{-1}]{\lcrxx[#1] \lceil {#2} \rceil}
+\newcommand{\norm}[2][{-1}]{\lcrxx[#1] \Vert {#2} \Vert}
+\newcommand{\inner}[3][{-1}]{\lcrxx[#1] \langle {{#2},\ {#3}} \rangle}
+\newcommand{\card}[2][{-1}]{\lcrxx[#1] \vert {#2} \vert }
```
The code looks right, and the examples work correctly. Not sure if \abs ... \card have the intended default behaviour at the moment

jnegrea reviewed (commented):
It works as it should.

jhuggins commented:
Fixed issues pointed out my @jnegrea as well as other improvements
* fixed incorrect defaults
* `\lrcx` -> `\@lrcx`
* `\lrcxx` -> `\lrcx`
* added Lp spaces, norms, and operators
* improved examples to be more comprehensive and put in to tables

trevorcampbell commented:
@jhuggins this is on my queue to review -- will get to it asap!

jnegrea commented:
changes look good. 
1. what is the meaning of the @ sign in the \@lcrx macro? is it just a name for the "helper function" to lcrx? or does it have some semantic meaning, or special functionality that i am missing?

2. the naming convention of using "arg" for versions of commands with and extra argument (e.g.   "\normarg", "\innerarg", "Lparg"...) seems more syntactic than semantic to me. it doesn't tell the reader what the extra argument *means*. If these didn't use bracket functions with optional  argument sizing then I'd suggest not having the arg versions and making the value of arg be an optional argument. Having multiple optional arguments in latex is possible, but leads to unintuitive behaviour.  Not sure what the best course is for these, but for Lp spaces at least it might be safe to assume that no resiszing will be needed for the name of the measure or that default rbra behaviour will be appropriate?

jhuggins commented:
> 1. what is the meaning of the @ sign in the `@lcrx` macro? is it just a name for the "helper function" to lcrx? or does it have some semantic meaning, or special functionality that i am missing?

Yeah, just a helper function. My understanding is that the @ sign is often used for "protected" commands/variables that a user normally wouldn't access. 

> 2. the naming convention of using "arg" for versions of commands with and extra argument (e.g. "\normarg", "\innerarg", "Lparg"...) seems more syntactic than semantic to me. it doesn't tell the reader what the extra argument means. If these didn't use bracket functions with optional argument sizing then I'd suggest not having the arg versions and making the value of arg be an optional argument. Having multiple optional arguments in latex is possible, but leads to unintuitive behaviour. Not sure what the best course is for these, but for Lp spaces at least it might be safe to assume that no resizing will be needed for the name of the measure or that default rbra behaviour will be appropriate?

I'm open to given these more semantically meaningful names. Maybe `\normsub` or `\normtype` with matching `\inner` naming? And `Lpmeas`?

trevorcampbell commented:
> I'm open to given these more semantically meaningful names. Maybe \normsub or \normtype with matching \inner naming? And Lpmeas?

Yeah, names derived from meaning are generally better I'd say. 

Probably a dumb question -- why do we need a special version of these commands to add a subscript? Can we not just do `\norm{blah}_2` for example?

Aside from that, I'd like to see more comprehensive documentation / examples in `examples.tex` (e.g., showing what the different sizes look like with different number options). After that I'd be happy to merge.

As a side note: I think I'd like to keep the earlier shortcuts `\pp(`, `\pp[` and `\cppl`. I think I personally would be more likely to use these in most cases because the command actually contains the symbol that will be shown (e.g. `(` for `\pp(` ) making the TeX look more like the math.
